File: 1988-minimize-maximum-pair-sum-in-array/minimize-maximum-pair-sum-in-array.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        maxi, left, right = 0, 0, len(nums)-1
        while left < right:
            maxi = max(maxi, nums[left] + nums[right])
            left += 1
            right -= 1
        return maxi

File: 1988-minimize-maximum-pair-sum-in-array/minimize-maximum-pair-sum-in-array.java
class Solution {
    public int minPairSum(int[] nums) {
        Arrays.sort(nums);
        int max = 0, left = 0, right = nums.length-1;
        while(left < right) max = Math.max(max, nums[left++] + nums[right--]);
        return max;
    }
}